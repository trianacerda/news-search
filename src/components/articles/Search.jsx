import React from 'react';
import PropTypes from 'prop-types';

export default function Search({ title, onTitleChange, onSubmit }) {
  return (
    <form onSubmit={onSubmit}>
      <label htmlFor="title">Article Title</label>
      <input
        id="title"
        title="title"
        type="text"
        value={title}
        onChange={onTitleChange}
      />
      <button aria-label="get-articles">Give Me Articles!</button>
    </form>
  );
}

Search.propTypes = {
  title: PropTypes.string.isRequired,
  onTitleChange: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
};
