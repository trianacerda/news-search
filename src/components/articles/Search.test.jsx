import React, { Component } from 'react';
import { render } from '@testing-library/react';
import Search from './Search';

describe('Search query test', () => {
  it('taking in a query and change on submite', () => {
    const { asFragment } = render(
      <Search onSubmit="dog" value="dog" onChange="dog" />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
