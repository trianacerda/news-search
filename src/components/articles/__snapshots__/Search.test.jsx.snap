// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search query test taking in a query and change on submite 1`] = `
<DocumentFragment>
  <form>
    <label
      for="title"
    >
      Article Title
    </label>
    <input
      id="title"
      title="title"
      type="text"
      value=""
    />
    <button
      aria-label="get-articles"
    >
      Give Me Articles!
    </button>
  </form>
</DocumentFragment>
`;
